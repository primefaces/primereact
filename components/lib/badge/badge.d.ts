/**
 *
 * Badge represents people using icons, labels and images.
 *
 * [Live Demo](https://www.primereact.org/badge)
 *
 * @module badge
 *
 */
import * as React from 'react';
import { ComponentHooks } from '../componentbase/componentbase';
import { PassThroughOptions } from '../passthrough';
import { PassThroughType } from '../utils/utils';

export declare type BadgePassThroughType<T> = PassThroughType<T, BadgePassThroughMethodOptions>;

/**
 * Custom passthrough(pt) option method.
 */
export interface BadgePassThroughMethodOptions {
    props: BadgeProps;
}

/**
 * Custom passthrough(pt) options.
 * @see {@link BadgeProps.pt}
 */
export interface BadgePassThroughOptions {
    /**
     * Uses to pass attributes to the root's DOM element.
     */
    root?: BadgePassThroughType<React.HTMLAttributes<HTMLSpanElement>>;
    /**
     * Used to manage all lifecycle hooks
     * @see {@link ComponentHooks}
     */
    hooks?: ComponentHooks;
}

/**
 * Defines valid properties in Badge component. In addition to these, all properties of HTMLSpanElement can be used in this component.
 * @group Properties
 */
export interface BadgeProps extends Omit<React.DetailedHTMLProps<React.HTMLAttributes<HTMLSpanElement>, HTMLSpanElement>, 'ref'> {
    /**
     * Value to display inside the badge.
     *
     */
    value?: any | null | undefined;
    /**
     * Severity type of the badge.
     * @defaultValue null
     */
    severity?: 'success' | 'info' | 'warning' | 'danger' | null | undefined;
    /**
     * Size of the badge, valid options are "large" and "xlarge".
     * @defaultValue null
     */
    size?: 'normal' | 'large' | 'xlarge' | null | undefined;
    /**
     * Used to get the child elements of the component.
     * @readonly
     */
    children?: React.ReactNode | undefined;
    /**
     * Uses to pass attributes to DOM elements inside the component.
     * @type {BadgePassThroughOptions}
     */
    pt?: BadgePassThroughOptions;
    /**
     * Used to configure passthrough(pt) options of the component.
     * @type {PassThroughOptions}
     */
    ptOptions?: PassThroughOptions;
    /**
     * When enabled, it removes component related styles in the core.
     * @defaultValue false
     */
    unstyled?: boolean;
}

/**
 * **PrimeReact - Badge**
 *
 * _Badge represents people using icons, labels and images._
 *
 * [Live Demo](https://www.primereact.org/badge/)
 * --- ---
 * ![PrimeReact](https://primefaces.org/cdn/primereact/images/logo-100.png)
 *
 * @group Component
 */
export declare class Badge extends React.Component<BadgeProps, any> {
    /**
     * Used to get container element.
     * @return {HTMLSpanElement} Container element
     */
    public getElement(): HTMLSpanElement;
}
