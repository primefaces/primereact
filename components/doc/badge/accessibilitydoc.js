import { CodeHighlight } from '../common/codehighlight';
import { DevelopmentSection } from '../common/developmentsection';
import { DocSubSection } from '../common/docsubsection';

export function AccessibilityDoc() {
    return (
        <>
            <DocSubSection id="accessibility" label="Accessibility">
                <DevelopmentSection>
                    <h4>Screen Reader</h4>
                    <p>
                        Badge does not include any roles and attributes by default, any attribute is passed to the root element so aria roles and attributes can be added if required. If the badges are dynamic,
                        <i>aria-live</i> may be utilized as well. In case badges need to be tabbable, <i>tabIndex</i> can be added to implement custom key handlers.
                    </p>

                    <h5>Keyboard Support</h5>
                    <p>Component does not include any interactive elements.</p>
                </DevelopmentSection>
            </DocSubSection>
        </>
    );
}
