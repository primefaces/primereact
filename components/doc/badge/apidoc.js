import Link from 'next/link';
import { CodeHighlight } from '../common/codehighlight';
import { DevelopmentSection } from '../common/developmentsection';
import { DocSectionText } from '../common/docsectiontext';
import { DocSubSection } from '../common/docsubsection';

export function ApiDoc(props) {
    return (
        <>
            <DocSectionText {...props}></DocSectionText>

            <DocSubSection id="severities" label="Severities">
                <p>
                    Different color options are available as severity levels. When used as a component use the <i>severity</i> property to apply a severity.
                </p>

                <ul>
                    <li>success</li>
                    <li>info</li>
                    <li>warning</li>
                    <li>danger</li>
                </ul>
                <CodeHighlight>
                    {`
<Badge value="2" severity="success"></Badge>
`}
                </CodeHighlight>
            </DocSubSection>
            <DocSubSection id="properties" label="Properties">
                <p>Any valid attribute is passed to the root element implicitly, extended properties are as follows;</p>
                <div className="doc-tablewrapper">
                    <table className="doc-table">
                        <thead>
                            <tr>
                                <th>Name</th>
                                <th>Type</th>
                                <th>Default</th>
                                <th>Description</th>
                            </tr>
                        </thead>
                        <tbody>
                            <tr>
                                <td>value</td>
                                <td>any</td>
                                <td>null</td>
                                <td>Value to display inside the badge.</td>
                            </tr>
                            <tr>
                                <td>severity</td>
                                <td>string</td>
                                <td>null</td>
                                <td>Severity type of the badge.</td>
                            </tr>
                            <tr>
                                <td>size</td>
                                <td>string</td>
                                <td>null</td>
                                <td>Size of the badge, valid options are "large" and "xlarge".</td>
                            </tr>
                        </tbody>
                    </table>
                </div>
            </DocSubSection>
        </>
    );
}
