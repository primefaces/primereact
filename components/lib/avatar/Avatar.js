import React, { forwardRef } from 'react';
import PropTypes from 'prop-types';
import { ObjectUtils, classNames, IconUtils } from '../utils/Utils';

export const Avatar = forwardRef((props, ref) => {

    const createContent = () => {
        if (props.label) {
            return <span className="p-avatar-text">{props.label}</span>
        }
        else if (props.icon) {
            return IconUtils.getJSXIcon(props.icon, { className: 'p-avatar-icon' }, { props });
        }
        else if (props.image) {
            return <img src={props.image} alt={props.imageAlt} onError={props.onImageError}></img>
        }

        return null;
    }

    const containerClassName = classNames('p-avatar p-component', {
        'p-avatar-image': props.image != null,
        'p-avatar-circle': props.shape === 'circle',
        'p-avatar-lg': props.size === 'large',
        'p-avatar-xl': props.size === 'xlarge',
        'p-avatar-clickable': !!props.onClick
    }, props.className);

    const content = props.template ? ObjectUtils.getJSXElement(props.template, props) : createContent();

    return (
        <div className={containerClassName} style={props.style} onClick={props.onClick}>
            {content}
            {props.children}
        </div>
    )
});

Avatar.defaultProps = {
    __TYPE: 'Avatar',
    label: null,
    icon: null,
    image: null,
    size: 'normal',
    shape: 'square',
    style: null,
    className: null,
    template: null,
    imageAlt: 'avatar',
    onImageError: null,
    onClick: null
}

Avatar.propTypes /* remove-proptypes */ = {
    __TYPE: PropTypes.string,
    label: PropTypes.string,
    icon: PropTypes.any,
    image: PropTypes.string,
    size: PropTypes.string,
    shape: PropTypes.string,
    style: PropTypes.object,
    className: PropTypes.string,
    template: PropTypes.any,
    imageAlt: PropTypes.string,
    onImageError: PropTypes.func,
    onClick: PropTypes.func
}
