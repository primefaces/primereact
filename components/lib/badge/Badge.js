import React, { forwardRef, memo } from 'react';
import PropTypes from 'prop-types';
import { classNames } from '../utils/Utils';

export const Badge = memo(forwardRef((props, ref) => {
    const className = classNames('p-badge p-component', {
        'p-badge-no-gutter': props.value && String(props.value).length === 1,
        'p-badge-dot': !props.value,
        'p-badge-lg': props.size === 'large',
        'p-badge-xl': props.size === 'xlarge',
        [`p-badge-${props.severity}`]: props.severity !== null
    }, props.className);

    return (
        <span className={className} style={props.style}>
            {props.value}
        </span>
    )
}));

Badge.defaultProps = {
    __TYPE: 'Badge',
    value: null,
    severity: null,
    size: null,
    style: null,
    className: null
}

Badge.propTypes /* remove-proptypes */ = {
    __TYPE: PropTypes.string,
    value: PropTypes.any,
    severity: PropTypes.string,
    size: PropTypes.string,
    style: PropTypes.object,
    className: PropTypes.string
}
