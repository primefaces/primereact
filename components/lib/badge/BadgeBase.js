import { ObjectUtils } from '../utils/Utils';

export const BadgeBase = {
    defaultProps: {
        __TYPE: 'Badge',
        value: null,
        severity: null,
        size: null,
        style: null,
        className: null,
        children: undefined
    },
    getProps: (props) => ObjectUtils.getMergedProps(props, BadgeBase.defaultProps),
    getOtherProps: (props) => ObjectUtils.getDiffProps(props, BadgeBase.defaultProps)
};
