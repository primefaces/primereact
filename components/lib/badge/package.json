{
    "main": "./badge.cjs.js",
    "module": "./badge.esm.js",
    "unpkg": "./badge.min.js",
    "types": "./badge.d.ts",
    "sideEffects": false
}
