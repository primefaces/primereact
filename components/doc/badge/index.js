import React, { memo } from 'react';
import Link from 'next/link';
import { TabView, TabPanel } from '../../lib/tabview/TabView';
import { useLiveEditorTabs } from '../common/liveeditor';
import { CodeHighlight } from '../common/codehighlight';

const BadgeDoc = memo(() => {

    const sources = {
        'class': {
            tabName: 'Class Source',
            content: `
import React, { Component } from 'react';
import { Button } from 'primereact/button';
import { Badge } from 'primereact/badge';

export class BadgeDemo extends Component {

    render() {
        return (
            <div>
                <div className="card">
                    <h5>Numbers</h5>
                    <Badge value="2" className="mr-2"></Badge>
                    <Badge value="8" severity="success" className="mr-2"></Badge>
                    <Badge value="4" severity="info" className="mr-2"></Badge >
                    <Badge value="12" severity="warning" className="mr-2"></Badge>
                    <Badge value="3" severity="danger"></Badge>

                    <h5 className="mb-4">Positioned Badge</h5>
                    <i className="pi pi-bell mr-4 p-text-secondary p-overlay-badge" style={{ fontSize: '2rem' }}><Badge value="2" ></Badge></i>
                    <i className="pi pi-calendar mr-4 p-text-secondary p-overlay-badge" style={{ fontSize: '2rem' }}><Badge value="10+" severity="danger" ></Badge></i>
                    <i className="pi pi-envelope p-text-secondary p-overlay-badge" style={{ fontSize: '2rem' }}><Badge severity="danger"></Badge></i>

                    <h5>Button Badge</h5>
                    <Button type="button" label="Emails" className="mr-2"><Badge value="8" ></Badge></Button>
                    <Button type="button" label="Messages" icon="pi pi-users" className="p-button-warning"><Badge value="8" severity="danger" ></Badge></Button>

                    <h5>Sizes</h5>
                    <Badge value="2" className="mr-2"></Badge>
                    <Badge value="4" className="mr-2" size="large" severity="warning"></Badge>
                    <Badge value="6" size="xlarge" severity="success"></Badge>
                </div>
            </div>
        );
    }
}
`
        },
        'hooks': {
            tabName: 'Hooks Source',
            content: `
import React from 'react';
import { Button } from 'primereact/button';
import { Badge } from 'primereact/badge';

export const BadgeDemo = () => {

    return (
        <div>
            <div className="card">
                <h5>Numbers</h5>
                <Badge value="2" className="mr-2"></Badge>
                <Badge value="8" severity="success" className="mr-2"></Badge>
                <Badge value="4" severity="info" className="mr-2"></Badge >
                <Badge value="12" severity="warning" className="mr-2"></Badge>
                <Badge value="3" severity="danger"></Badge>

                <h5 className="mb-4">Positioned Badge</h5>
                <i className="pi pi-bell mr-4 p-text-secondary p-overlay-badge" style={{ fontSize: '2rem' }}><Badge value="2" ></Badge></i>
                <i className="pi pi-calendar mr-4 p-text-secondary p-overlay-badge" style={{ fontSize: '2rem' }}><Badge value="10+" severity="danger" ></Badge></i>
                <i className="pi pi-envelope p-text-secondary p-overlay-badge" style={{ fontSize: '2rem' }}><Badge severity="danger"></Badge></i>

                <h5>Button Badge</h5>
                <Button type="button" label="Emails" className="mr-2"><Badge value="8" ></Badge></Button>
                <Button type="button" label="Messages" icon="pi pi-users" className="p-button-warning"><Badge value="8" severity="danger" ></Badge></Button>

                <h5>Sizes</h5>
                <Badge value="2" className="mr-2"></Badge>
                <Badge value="4" className="mr-2" size="large" severity="warning"></Badge>
                <Badge value="6" size="xlarge" severity="success"></Badge>
            </div>
        </div>
    );
}
`
        },
        'ts': {
            tabName: 'TS Source',
            content: `
import React from 'react';
import { Button } from 'primereact/button';
import { Badge } from 'primereact/badge';

export const BadgeDemo = () => {

    return (
        <div>
            <div className="card">
                <h5>Numbers</h5>
                <Badge value="2" className="mr-2"></Badge>
                <Badge value="8" severity="success" className="mr-2"></Badge>
                <Badge value="4" severity="info" className="mr-2"></Badge >
                <Badge value="12" severity="warning" className="mr-2"></Badge>
                <Badge value="3" severity="danger"></Badge>

                <h5 className="mb-4">Positioned Badge</h5>
                <i className="pi pi-bell mr-4 p-text-secondary p-overlay-badge" style={{ fontSize: '2rem' }}><Badge value="2" ></Badge></i>
                <i className="pi pi-calendar mr-4 p-text-secondary p-overlay-badge" style={{ fontSize: '2rem' }}><Badge value="10+" severity="danger" ></Badge></i>
                <i className="pi pi-envelope p-text-secondary p-overlay-badge" style={{ fontSize: '2rem' }}><Badge severity="danger"></Badge></i>

                <h5>Button Badge</h5>
                <Button type="button" label="Emails" className="mr-2"><Badge value="8" ></Badge></Button>
                <Button type="button" label="Messages" icon="pi pi-users" className="p-button-warning"><Badge value="8" severity="danger" ></Badge></Button>

                <h5>Sizes</h5>
                <Badge value="2" className="mr-2"></Badge>
                <Badge value="4" className="mr-2" size="large" severity="warning"></Badge>
                <Badge value="6" size="xlarge" severity="success"></Badge>
            </div>
        </div>
    );
}
`
        },
        'browser': {
            tabName: 'Browser Source',
            imports: `
        <script src="https://unpkg.com/primereact/api/api.min.js"></script>
        <script src="https://unpkg.com/primereact/core/core.min.js"></script>
        <script src="https://unpkg.com/primereact/button/button.min.js"></script>
        <script src="https://unpkg.com/primereact/badge/badge.min.js"></script>`,
            content: `
const { Button } = primereact.button;
const { Badge } = primereact.badge;

const BadgeDemo = () => {

    return (
        <div>
            <div className="card">
                <h5>Numbers</h5>
                <Badge value="2" className="mr-2"></Badge>
                <Badge value="8" severity="success" className="mr-2"></Badge>
                <Badge value="4" severity="info" className="mr-2"></Badge >
                <Badge value="12" severity="warning" className="mr-2"></Badge>
                <Badge value="3" severity="danger"></Badge>

                <h5 className="mb-4">Positioned Badge</h5>
                <i className="pi pi-bell mr-4 p-text-secondary p-overlay-badge" style={{ fontSize: '2rem' }}><Badge value="2" ></Badge></i>
                <i className="pi pi-calendar mr-4 p-text-secondary p-overlay-badge" style={{ fontSize: '2rem' }}><Badge value="10+" severity="danger" ></Badge></i>
                <i className="pi pi-envelope p-text-secondary p-overlay-badge" style={{ fontSize: '2rem' }}><Badge severity="danger"></Badge></i>

                <h5>Button Badge</h5>
                <Button type="button" label="Emails" className="mr-2"><Badge value="8" ></Badge></Button>
                <Button type="button" label="Messages" icon="pi pi-users" className="p-button-warning"><Badge value="8" severity="danger" ></Badge></Button>

                <h5>Sizes</h5>
                <Badge value="2" className="mr-2"></Badge>
                <Badge value="4" className="mr-2" size="large" severity="warning"></Badge>
                <Badge value="6" size="xlarge" severity="success"></Badge>
            </div>
        </div>
    );
}
                `
        }
    }

    return (
        <div className="content-section documentation" id="app-doc">
            <TabView>
                <TabPanel header="Documentation">
                    <h5>Getting Started</h5>
                    <p>Badge can be used as a component.</p>

                    <h6>Component</h6>
<CodeHighlight lang="js">
{`
import { Badge } from 'primereact/badge';
`}
</CodeHighlight>
                    <p>Content of the badge is specified using the <i>value</i> property.</p>
<CodeHighlight>
{`
<Badge value="2"></Badge>
`}
</CodeHighlight>
                    <h5>Severities</h5>
                    <p>Different color options are available as severity levels. When used as a component use the <i>severity</i> property
                    to apply a severity.</p>

                    <ul>
                        <li>success</li>
                        <li>info</li>
                        <li>warning</li>
                        <li>danger</li>
                    </ul>
<CodeHighlight>
{`
<Badge value="2" severity="success"></Badge>
`}
</CodeHighlight>

                    <h5>Button Badges</h5>
                    <p>Buttons provide integrated badge support with the <i>badge</i> and <i>badgeClass</i> properties.</p>

<CodeHighlight>
{`
<Button type="button" label="Emails">
    <Badge value="2"></Badge>
</Button>

<Button type="button" label="Messages" icon="pi pi-users" className="p-button-warning">
    <Badge value="8" severity="danger">
</Button>
`}
</CodeHighlight>
                    <h5>Sizes</h5>
                    <p>Badge sizes are adjusted with the <i>size</i> property that accepts "large" and "xlarge" as the possible alternatives to the default size.</p>
<CodeHighlight>
{`
<Badge value="2"></Badge>
<Badge value="4" size="large" severity="warning"></Badge>
<Badge value="6" size="xlarge" severity="success"></Badge>
`}
</CodeHighlight>
                    <p>In addition, when placed inside another element, badge sizes can also derive their size from their parent.</p>
<CodeHighlight>
{`
<h1>Heading 1 <Badge value="New"></Badge></h1>
<h2>Heading 2 <Badge value="New"></Badge></h2>
`}
</CodeHighlight>
                    <h5>Properties</h5>
                    <p>Any property as style and class are passed to the main container element. Following are the additional properties to configure the component.</p>
                    <div className="doc-tablewrapper">
                        <table className="doc-table">
                            <thead>
                                <tr>
                                    <th>Name</th>
                                    <th>Type</th>
                                    <th>Default</th>
                                    <th>Description</th>
                                </tr>
                            </thead>
                            <tbody>
                                <tr>
                                    <td>value</td>
                                    <td>any</td>
                                    <td>null</td>
                                    <td>Value to display inside the badge.</td>
                                </tr>
                                <tr>
                                    <td>severity</td>
                                    <td>string</td>
                                    <td>null</td>
                                    <td>Severity type of the badge.</td>
                                </tr>
                                <tr>
                                    <td>size</td>
                                    <td>string</td>
                                    <td>null</td>
                                    <td>Size of the badge, valid options are "large" and "xlarge".</td>
                                </tr>
                            </tbody>
                        </table>
                    </div>

                    <h5>Styling</h5>
                    <p>Following is the list of structural style classes, for theming classes visit <Link href="/theming">theming</Link> page.</p>
                    <div className="doc-tablewrapper">
                        <table className="doc-table">
                            <thead>
                            <tr>
                                <th>Name</th>
                                <th>Element</th>
                            </tr>
                            </thead>
                            <tbody>
                                <tr>
                                    <td>p-badge</td>
                                    <td>Badge element</td>
                                </tr>
                                <tr>
                                    <td>p-overlay-badge</td>
                                    <td>Wrapper of a badge and its target.</td>
                                </tr>
                                <tr>
                                    <td>p-badge-dot</td>
                                    <td>Badge element with no value.</td>
                                </tr>
                                <tr>
                                    <td>p-badge-success</td>
                                    <td>Badge element with success severity.</td>
                                </tr>
                                <tr>
                                    <td>p-badge-info</td>
                                    <td>Badge element with info severity.</td>
                                </tr>
                                <tr>
                                    <td>p-badge-warning</td>
                                    <td>Badge element with warning severity.</td>
                                </tr>
                                <tr>
                                    <td>p-badge-danger</td>
                                    <td>Badge element with danger severity.</td>
                                </tr>
                                <tr>
                                    <td>p-badge-lg</td>
                                    <td>Large badge element</td>
                                </tr>
                                <tr>
                                    <td>p-badge-xl</td>
                                    <td>Extra large badge element</td>
                                </tr>
                            </tbody>
                        </table>
                    </div>

                    <h5>Dependencies</h5>
                    <p>None.</p>
                </TabPanel>

                {
                    useLiveEditorTabs({ name: 'BadgeDemo', sources: sources })
                }
            </TabView>
        </div>
    );
})

export default BadgeDoc;
