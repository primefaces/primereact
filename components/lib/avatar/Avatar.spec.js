import '@testing-library/jest-dom';
import { Badge } from '../badge/Badge';
import { Avatar } from './Avatar';

import { snapshot } from '../../test';

describe('Avatar', () => {
    snapshot(<Avatar />, 'default');
    snapshot(<Avatar label="P" />, 'label');
    snapshot(<Avatar icon="pi pi-search" />, 'icon');
    snapshot(<Avatar image="user.png" />, 'image');
    snapshot(<Avatar size="large" />, 'size large');
    snapshot(<Avatar size="xlarge" />, 'size xlarge');
    snapshot(<Avatar shape="square" />, 'shape square');
    snapshot(<Avatar size="cirle" />, 'shape cirle');
    snapshot(<Avatar>Content</Avatar>, 'templating');
    snapshot(
        <Avatar image="user.png" size="xlarge">
            <Badge value="4" severity="danger" />
        </Avatar>,
        'badge'
    );
});
