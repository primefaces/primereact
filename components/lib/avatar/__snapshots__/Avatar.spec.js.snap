// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar badge 1`] = `
<div>
  <div
    class="p-avatar p-component p-avatar-image p-avatar-xl"
  >
    <img
      alt="avatar"
      src="user.png"
    />
    <span
      class="p-badge p-component p-badge-no-gutter p-badge-danger"
    >
      4
    </span>
  </div>
</div>
`;

exports[`Avatar default 1`] = `
<div>
  <div
    class="p-avatar p-component"
  />
</div>
`;

exports[`Avatar icon 1`] = `
<div>
  <div
    class="p-avatar p-component"
  >
    <span
      class="p-avatar-icon pi pi-search"
    />
  </div>
</div>
`;

exports[`Avatar image 1`] = `
<div>
  <div
    class="p-avatar p-component p-avatar-image"
  >
    <img
      alt="avatar"
      src="user.png"
    />
  </div>
</div>
`;

exports[`Avatar label 1`] = `
<div>
  <div
    class="p-avatar p-component"
  >
    <span
      class="p-avatar-text"
    >
      P
    </span>
  </div>
</div>
`;

exports[`Avatar shape cirle 1`] = `
<div>
  <div
    class="p-avatar p-component"
  />
</div>
`;

exports[`Avatar shape square 1`] = `
<div>
  <div
    class="p-avatar p-component"
  />
</div>
`;

exports[`Avatar size large 1`] = `
<div>
  <div
    class="p-avatar p-component p-avatar-lg"
  />
</div>
`;

exports[`Avatar size xlarge 1`] = `
<div>
  <div
    class="p-avatar p-component p-avatar-xl"
  />
</div>
`;

exports[`Avatar templating 1`] = `
<div>
  <div
    class="p-avatar p-component"
  >
    Content
  </div>
</div>
`;
